ci/Dockerfile-envoy

```diff
@@ -17,8 +17,9 @@ FROM ${BUILD_OS}:${BUILD_TAG} AS envoy-base
 ENV DEBIAN_FRONTEND=noninteractive
 EXPOSE 10000
 CMD ["envoy", "-c", "/etc/envoy/envoy.yaml"]
+# Ensure the envoy user is able to write to container logs owned by root:tty
 RUN mkdir -p /etc/envoy \
-    && adduser --group --system envoy
+    && useradd --system --create-home --groups tty --shell /usr/sbin/nologin
```

Comment:
i agree that `useradd` is generally a better choice for scripts/etc but i think this has broken due to the `--system` flag

iirc it sets the user id from a different range to that which is used by real users, and sets no home directory

i think we want something like:

```console
useradd -u 999 --no-create-home --groups tty --shell /usr/sbin/nologin
```

Reply:
also we need to define the user - so actually more like ...

```console
useradd -u 999 --no-create-home --groups tty --shell /usr/sbin/nologin envoy
```

Reply:
hmm - it might just be that the problem was the missing username - if that is the case no need to define shell/uid/no-create-home etc - `--system` should probably take care of it

Reply:
testing further i think we _do_ want some of the other flags - this i think

```console
useradd --system --no-create-home -d /nonexistent --groups tty --shell /usr/sbin/nologin envoy
```

Reply:
The original adduser was creating a home directory so I kept the --create-home option for useradd since as long as the shell is a non-shell, logging in (like ssh) will not be possible. 
```
root@1d083228ee61:/# useradd --system --create-home --groups tty --shell /usr/sbin/nologin  envoy3
root@1d083228ee61:/# id envoy3
uid=998(envoy3) gid=998(envoy3) groups=998(envoy3),5(tty)
root@1d083228ee61:/# cat /etc/passwd | grep envoy
envoy:x:101:101::/home/envoy:/usr/sbin/nologin               <- adduser
envoy3:x:998:998::/home/envoy3:/usr/sbin/nologin        <- useradd
```
I adjusted the script based on your suggestion. Thanks!